tests/test_generichash.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2016 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+import binascii
+import json
+import os
+
+import pytest
+
+import nacl.encoding
+import nacl.exceptions as exc
+import nacl.hash
+from nacl.utils import bytes_as_string
+
+OVERLONG_PARAMS_VECTORS = [
+    (b'key', 65*b'\xaa', 16*b'\xaa', 16*b'\x55', 64, b'will raise'),
+    (b'salt', b'key', 17*b'\xaa', 16*b'\x55', 64, b'will raise'),
+    (b'personal', b'key', 16*b'\xaa', 17*b'\x55', 64, b'will raise'),
+    (b'digest_size', b'key', 16*b'\xaa', 16*b'\x55', 65, b'will raise'),
+]
+
+
+def _read_crypto_test_vectors(fname):
+    vectors = []
+    path = os.path.join(os.path.dirname(__file__), "data", fname)
+    with open(path, "r") as fp:
```

Comment:
You can leave it here and we can discuss whether it needs to exist in the init/update/finalize PR :)